LoopKitUI/Views/Settings Editors/TherapySettingsViewModel.swift

```diff
@@ -111,3 +116,58 @@ public class TherapySettingsViewModel: ObservableObject {
         didSave?(TherapySetting.insulinSensitivity, therapySettings)
     }
 }
+
+// MARK: Navigation
+
+extension TherapySettingsViewModel {
+
+    func screen(for setting: TherapySetting) -> (_ goBack: @escaping () -> Void) -> AnyView {
+        switch setting {
+        case .suspendThreshold:
+            return { goBack in
+                AnyView(SuspendThresholdEditor(therapySettingsViewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .glucoseTargetRange:
+            return { goBack in
+                AnyView(CorrectionRangeScheduleEditor(viewModel: self, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .preMealCorrectionRangeOverride:
+            return { goBack in
+                AnyView(CorrectionRangeOverridesEditor(viewModel: self, preset: .preMeal, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .workoutCorrectionRangeOverride:
+            return { goBack in
+                AnyView(CorrectionRangeOverridesEditor(viewModel: self, preset: .workout, didSave: goBack).environment(\.dismiss, goBack))
+            }
+        case .basalRate:
+            if self.pumpSupportedIncrements?() != nil {
```

Comment:
[nit] So, in this case it displays the default value below? Perhaps this check should really be a precondition?